packages/core/src/dispatcher.ts

```diff
@@ -360,7 +362,10 @@ export class Dispatcher {
               signal: signal,
             })
           )
-          blockCid = resolution.cid
+          // HACK around ipfs-core-types multiformats version mismatch
+          // don't just as CID because we want to new version with the readonly field rather than getter
+          // that is dropped during a deepClone... assuming it actually matters...
+          blockCid = toCID(resolution.cid.toString())
```

Comment:
this makes me pretty sad.  Your comment implies it might not be necessary?  Can we test that and remove the hack if it's not needed?

Reply:
Or can we replace our use of ipfs-core-types with using the multiformats version of cids instead?

Reply:
yeah, I agree it's awful. I tried to cover this in the summary cause it's weird. And the comment (and multiple versions don't read _super_ clean).

So it didn't seem to matter when I just used `old as CID` to make typescript happy. I _think_ it's technically different, as it's 2 js classes, one with a getter and one with a readonly field, but otherwise they should be about the same (v11 vs v13). That's what I was trying to get at with "actually matters". Cause even the mismatch equals errors we get don't _actually_ matter, as everything is identical except a field definition, but node still seems to find the string info correctly.

ipfs-core-types is deprecated in favor of helia and uses multiformats v11. I didn't want to update to helia (assumed it'd be hard without looking). We could provide a thin wrapper around the interface with our own that uses a v13 CID since we know they should be compatible, but I'm not sure if it's worth it. 

anyway, let me know what you think.

Reply:
Yeah, my understanding of all the work that's happened in the last 2 days has been around trying to make sure we only use v13 CIDs and remove every place that's pulling in older versions of the multiformats package.  So it seems like this is the last remaining holdout.  So it does seem like the right long term fix is to move away from `ipfs-core-types`.  But I also don't want to hold up this PR.  So I guess I'd say let's file a CORE ticket in linear to switch to helia for our ipfs http client, and add a `TODO(CORE-XXXX):` comment to each of the places we do this hack so we know to remove this when we make that upgrade.

Reply:
Comments updated to `// TODO(CORE-137) - Replace ipfs-core-types to get consistent multiformats version`